File: app/src/main/java/com/gasgasstation/base/BaseActivity.java
package com.gasgasstation.base;

import android.support.annotation.LayoutRes;
import android.support.v7.app.AppCompatActivity;

import butterknife.ButterKnife;

/**
 * Created by kws on 2017. 11. 1..
 */

public abstract class BaseActivity extends AppCompatActivity{

    protected void onPrepareActivity() {

    }

    protected void onPreparePresenter() {

    }

    @LayoutRes
    protected abstract int getLayoutResID();

    private void bindViews() {
        ButterKnife.bind(this);
    }
}

File: app/src/main/kotlin/com/gasgasstation/base/BaseActivity.kt
package com.gasgasstation.base

import android.os.Bundle
import android.support.annotation.LayoutRes
import android.support.v7.app.AppCompatActivity

/**
 * Created by kws on 2017. 11. 22..
 */
abstract class BaseActivity : AppCompatActivity() {
    @LayoutRes abstract fun getLayoutResId(): Int

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(getLayoutResId())
    }

}